app/ui-react/syndesis/src/modules/connections/components/ConnectionsWithToolbarNew.tsx

```diff
@@ -0,0 +1,126 @@
+import { Connection } from '@syndesis/models';
+import {
+  ConnectionsListView,
+  IActiveFilter,
+  IConnectionsListViewProps,
+  IFilterType,
+  ISortType,
+} from '@syndesis/ui';
+import { WithListViewToolbarHelpers } from '@syndesis/utils';
+import * as React from 'react';
+import { Translation } from 'react-i18next';
+import i18n from '../../../i18n';
+import resolvers from '../resolvers';
+import { Connections, IConnectionsProps } from './Connections';
+
+function getFilteredAndSortedConnections(
+  connections: Connection[],
+  activeFilters: IActiveFilter[],
+  currentSortType: ISortType,
+  isSortAscending: boolean
+) {
+  let filteredAndSortedConnections = connections;
+  activeFilters.forEach((filter: IActiveFilter) => {
+    const valueToLower = filter.value.toLowerCase();
+    filteredAndSortedConnections = filteredAndSortedConnections.filter(
+      (c: Connection) => c.name.toLowerCase().includes(valueToLower)
+    );
+  });
+
+  filteredAndSortedConnections = filteredAndSortedConnections.sort(
+    (miA, miB) => {
+      const left = isSortAscending ? miA : miB;
+      const right = isSortAscending ? miB : miA;
+      return left.name.localeCompare(right.name);
+    }
+  );
+
+  return filteredAndSortedConnections;
+}
+
+const filterByName = {
+  filterType: 'text',
+  id: 'name',
+  placeholder: i18n.t('shared:filterByNamePlaceholder'),
+  title: i18n.t('shared:Name'),
+} as IFilterType;
+
+const filterTypes = [filterByName];
+
+const sortByName = {
+  id: 'name',
+  isNumeric: false,
+  title: i18n.t('shared:Name'),
+} as ISortType;
+
+const sortTypes: ISortType[] = [sortByName];
+
+export interface IConnectionsWithToolbarNewProps
+  extends IConnectionsProps,
+    Pick<IConnectionsListViewProps, 'createConnectionButtonStyle'> {
+  children?: any;
+}
+
+export class ConnectionsWithToolbarNew extends React.Component<
```

Comment:
U need a better name.  I wonder, maybe this should live in the `integrations` module and maybe be called like I dunno, `AllStepsWithToolbar` or something?  That way it's a bit more obvious that this guy isn't just for "connections" in the syndesis sense.  Ooh ooh, maybe `UIStepsWithToolbar`

Reply:
Works great BTW!

Reply:
Yeah I was struggling a bit with the name, as you can see. 😂 I think `UIStepsWithToolbar` sounds good. 👍